modules/islandora_google_scholar/islandora_google_scholar.module

```diff
@@ -44,35 +43,45 @@ function islandora_google_scholar_create_meta_tags($object) {
     else {
       return FALSE;
     }
-    foreach ($mods_xml->xpath('mods:name') as $name_xml) {
+    foreach ($mods_xml->xpath(variable_get('islandora_scholar_xpaths_authors_xpath', 'mods:name')) as $name_xml) {
       $name_parts = array();
       $name_xml->registerXPathNamespace('mods', 'http://www.loc.gov/mods/v3');
-      $roles = $name_xml->xpath('mods:role/mods:roleTerm');
```

Comment:
Why did you remove the $role stuff? Identifying the author is important for this.

It looks like your version just take any names that it finds regardless of roleTerm? Is that correct? And is that deliberate?

Reply:
Because in the settings in Web UI, it specific ask for XPath to Author. We don't need to check again if it is labeled as Author. We could probably set a default XPath that will check for roleTerm. But if user has specified a XPath, we should use it as is.

Reply:
Ah, makes sense. 👍